oop/homework2.php

```diff
@@ -0,0 +1,65 @@
+<?php 
+abstract class Emailer 
+{
+    // Declare properties
+    protected $sender;
+    protected $recipients;
+    protected $subject;
+    protected $body;
+
+    // Use Constructor to initialize sender and recipients
+    function __construct($sender)
+    {
+        $this->sender = (object)$sender;
+        $this->recipients = array(); 
+    }
+
+    public function addRecipients($recipient)
+    {
+        array_push($this->recipients, $recipient);
+        return $this;
+    }
+
+    public function setSubject($subject)
+    {
+        $this->subject = $subject;
+        return $this;
+    }
+
+    public function setBody($body)
+    {
+        $this->body = $body;
+        return $this;
+    }
+}
+
+// Implement method sendEmail for SendGrid and MailChimp
+
+class SendGrid extends Emailer 
+{
+    public function sendEmail()
+    {
+        foreach ($this->recipients as $recipient) {
+            $result = mail($recipient, $this->subject, $this->body, "From: {$this->sender}\r\n");
+            echo "SendGrid successfully sent to {$recipient}\n";
+            echo "Sender: $this->sender\n";
+            echo "Subject: $this->subject\n";
+            echo "Content: $this->body\n";
+        }
+    }
+}
+
+class MailChimp extends Emailer 
+{
+    public function sendEmail()
+    {
+        foreach ($this->recipients as $recipient) {
+            $result = mail($recipient, $this->subject, $this->body, "From: {$this->sender}\r\n");
+            echo "MailChimp successfully sent to {$recipient}\n";
+            echo "Sender: $this->sender\n";
+            echo "Subject: $this->subject\n";
+            echo "Content: $this->body\n";
+        }
+    }
+}
```

Comment:
@ariga914 
I think need to have some sources to test these classes ?

Reply:
I've got it now.
I added some sentences for tests.